File: HrNewPortal.Models/Poll.cs
using System;

namespace HrNewsPortal.Models
{
    public class Poll
    {
        /// <summary>
        /// The item's unique identifier.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// The user name of the item's author.
        /// </summary>
        public string By { get; set; }

        /// <summary>
        /// Creation UTC date of the item.
        /// </summary>
        public DateTime? Time { get; set; }

        /// <summary>
        /// The comment story or poll text.
        /// </summary>
        public string Text { get; set; }

        /// <summary>
        /// The item's comments, in ranked display order.
        /// </summary>
        public Comment[] Kids { get; set; }

        /// <summary>
        /// The story's score, or the votes
        /// for a pollopt (poll option).
        /// </summary>
        public int Score { get; set; }

        /// <summary>
        /// The title of the story, poll, or job.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// A list of related Poll Options.
        /// </summary>
        /// <remarks>
        /// Come in display order.
        /// </remarks>
        public PollOption[] Parts { get; set; }

        /// <summary>
        /// In the case of stories or polls,
        /// the total comment count.
        /// </summary>
        public int Descendant { get; set; }
    }
}

File: HrNewsPortal.Portal/ClientApp/src/app/models/poll.ts
import { PollOption } from "./poll-option";
import { Comment } from "./comment";

export class Poll {
  // The item's unique identifier.
  public id: number;

  // The user name of the item's author.
  public by: string;

  // Creation UTC date of the item.
  public time?: Date;

  // The comment story or poll text.
  public text: string;

  // The identifiers of the item's
  // comments, in ranked display order.
  public kids: number[];

  // The comments, in ranked display order.
  public comments: Comment[];

  // The story's score, or the votes
  // for a pollopt (poll option).
  public score: number;

  // The title of the story, poll, or job.
  public title: string;

  /// A list of related Poll Options ids.
  /// Come in display order.
  public parts: number[];

  // A list of related Poll Options.
  // Come in display order.
  public pollOptions: PollOption[];

  // In the case of stories or polls,
  // the total comment count.
  public descendant: number;
}
